src/backend/src/services/work-packages.services.ts

```diff
@@ -729,4 +730,72 @@ export default class WorkPackagesService {
 
     return workPackageTemplateTransformer(workPackage);
   }
+
+  static async editWorkPackageTemplate(
+    user: User,
+    workPackageTemplateId: string,
+    templateName: string,
+    templateNotes: string,
+    duration: number | undefined,
+    stage: WorkPackageStage | undefined,
+    blockedBy: BlockedByInfo[],
+    expectedActivities: string[],
+    deliverables: string[],
+    workPackageName: string | undefined
+  ): Promise<Work_Package_Template> {
+    const originalWorkPackageTemplate = await prisma.work_Package_Template.findUnique({
+      where: { workPackageTemplateId },
+      include: { blockedBy: true }
+    });
+
+    if (!originalWorkPackageTemplate) throw new NotFoundException('Work Package Template', workPackageTemplateId);
+    if (originalWorkPackageTemplate.dateDeleted) throw new DeletedException('Work Package Template', workPackageTemplateId);
+    if (!isAdmin(user.role)) throw new AccessDeniedGuestException('edit work package templates');
+
+    const blockedByIds = originalWorkPackageTemplate.blockedBy.map((item) => item.blockedByInfoId);
+
+    if (blockedByIds.length > 0) {
+      await prisma.blocked_By_Info.deleteMany({
+        where: {
+          blockedByInfoId: {
+            in: blockedByIds
+          }
+        }
+      });
+    }
+
+    const isNewBlockedBy = (blockedByItem: BlockedByInfo) => {
+      return !originalWorkPackageTemplate.blockedBy.some(
+        (oldItem) => oldItem.blockedByInfoId === blockedByItem.blockedByInfoId
+      );
+    };
+
+    const blockedByToCreate = blockedBy.filter(isNewBlockedBy);
+
+    for (const blockedByItem of blockedByToCreate) {
```

Comment:
this looks good, but you also need to add code below this to update any blockedBys that are in both lists